code/core/src/telemetry/package-json.ts

```diff
@@ -12,18 +15,29 @@ export const getActualPackageVersion = async (packageName: string) => {
   try {
     const packageJson = await getActualPackageJson(packageName);
     return {
-      name: packageName,
-      version: packageJson.version,
+      name: packageJson?.name || packageName,
+      version: packageJson?.version || null,
     };
   } catch (err) {
-    return { name: packageName, version: null };
+    return {
+      name: packageName,
+      version: null,
+    };
   }
 };
 
-export const getActualPackageJson = async (packageName: string) => {
-  const resolvedPackageJson = require.resolve(join(packageName, 'package.json'), {
-    paths: [process.cwd()],
-  });
+export const getActualPackageJson = async (
+  packageName: string
+): Promise<PackageJson | undefined> => {
+  let resolvedPackageJson = await findUp('package.json', { cwd: require.resolve(packageName) });
```

Comment:
Do we know the perf implications of this?

Reply:
not noticeable I think. the function was already async.